### PR TITLE
Ft requesting #175743342

### PR DESCRIPTION
**Pull request by jemmyarina**
**What does this PR do?**  
User request an accommodation after logging in  
Add swagger api documentation  
Request status datatype updated from string to boolean

**Description of Task to be completed?**   
N/A

**How should this be manually tested?**  
Clone the repository.  
In your home directory move into the cloned folder by ft-requesting-#175743342  
Then run npm install to install all dependencies.  
Run npm run migrate to run migrations  
Run npm start to start the server.  

**Any background context you want to provide?**  
N/A  

**What are the relevant Pivotal Tracker stories?**  
#175743342

### CODE REVIEW

**Review by ConstantineMaliza — CHANGES_REQUESTED** [review]
Great work @jemmyarina  there is one thing remaining to cover all test

File: src/database/seeders/20210120052931-userRoles.js [hunk 1]
```diff
@@ -2,8 +2,12 @@
...
   up: async (queryInterface, Sequelize) => {
     await queryInterface.bulkInsert('userRoles', [{
-      name: process.env.ADMIN_ROLE_NAME,
-      description: process.env.ADMIN_ROLE_DESCRIPTION,
+      // name: process.env.ADMIN_ROLE_NAME,
+      // description: process.env.ADMIN_ROLE_DESCRIPTION,
+      // createdAt: new Date(),
+      // updatedAt: new Date()
+      name: 'MEMBER',
+      description: 'This is a member',
```
Comment (SimonImanigirimpuhwe) [inline src/database/seeders/20210120052931-userRoles.js]: Why changing this seeders?

File: src/controllers/requestController.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import jwt_decode from 'jwt-decode';
+
+// creating a request
+const createRequest = async (req, res) => {
+  try {
+    // const authHeader = req.headers.authorization.split(' ');
+    // const [authString, token] = authHeader;
+    // const decodedToken = jwt_decode(token);
+    // req.body.idUser=decodedToken.id;
+
```
Comment (SimonImanigirimpuhwe) [inline src/controllers/requestController.js]: Remove commented lines that you no longer need

File: src/routes/requestRoute.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+ *        properties:
+ *          dateStart:
+ *            type: string
+ *          dateEnd:
+ *            type: string
+ *          idRoom:
+ *             type: number
+ */
+router.post('/', protection, createR);
+
```
Comment (SimonImanigirimpuhwe) [inline src/routes/requestRoute.js]: Consider validation your request otherwise you can send multiple requests with the same data and also you can even send request with empty data
Screen Shot 2021-01-25 at 15 13 13
.

**Review by SimonImanigirimpuhwe — CHANGES_REQUESTED** [review]
Other from that you did great!

File: src/database/seeders/20210120052931-userRoles.js (hunk 1 above)
Comment (jemmyarina) [reply src/database/seeders/20210120052931-userRoles.js]: @SimonImanigirimpuhwe
Since you created **SUPER_ADMIN** Role in (src/database/adminRole.create.js), we do not need to seed it again.
Thanks for your concern!

**Review by furebo — CHANGES_REQUESTED** [review]
Please , remove the file src/tests/request.test2.js . It is useless because it is all commented.

**Review by SamuelSabasNayo — COMMENTED** [review]
@jemmyarina Please respond to the requested changes.

**Review by furebo — APPROVED** [review]

**Review by SimonImanigirimpuhwe — APPROVED** [review]

File: src/middlewares/check-auth.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ module.exports = (req, res, next) => {
         const token = req.headers.authorization.split(" ")[1];
         const decoded = jwt.verify(token, process.env.JWT_KEY);
         req.userData = decoded;
+        console.log(decoded)
```
Comment (SamuelSabasNayo) [inline src/middlewares/check-auth.js]: Please remove this console log.

File: src/services/roomService 2.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import models from '../database/models';
```
Comment (SamuelSabasNayo) [inline src/services/roomService 2.js]: Is this file needed?
Since we have **roomService.js**?

**Review by SamuelSabasNayo — CHANGES_REQUESTED** [review]

File: src/services/roomService 2.js (hunk 5 above)
Comment (SamuelSabasNayo) [inline src/services/roomService 2.js]: Is this file needed?
Since we have **roomService.js**?

**Review by SamuelSabasNayo — CHANGES_REQUESTED** [review]

**Review by SamuelSabasNayo — APPROVED** [review]

**Review by ConstantineMaliza — APPROVED** [review]

File: src/controllers/requestController.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+// creating a request
+const createRequest = async (req, res) => {
+  try {
+    const { idUser, idRoom, dateStart, dateEnd } = req.body,
+      { id } = req.userData;
+    const existingRoom = await findRooms(idRoom);
+
+    if(!existingRoom) return res.status(400).json({ message: 'Room id does not exist' });
+    console.log(existingRoom);
```
Comment (Anguandia) [inline src/controllers/requestController.js]: Please do not log to console outside your development environment. Check this and all instances of console.log()

File: src/controllers/requestController.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      dateStart,
+      dateEnd
+    };
+    const request = await model.request.create(newRequest);
+
+    console.log(idUser);
+  
+    return res.status(201).json({ message: 'Request created successfully!', request});
+  } catch (error) {
+    return res.status(500).json({error: error.message})
```
Comment (Anguandia) [inline src/controllers/requestController.js]: Please avoid returning technical error messages to the user, they will not understand. a simple message like 'internal error, contact support, etc is good for the user. Technical errors can be logged using a preferred logging service for maintenance and support. Check all instances where you return error.message in the response.
Additionally, since these technical error messages are not known beforehand, they will undermine the multi-language support feature

File: src/routes/requestRoute.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
+const controller = require('../controllers/requestController');
+const { Router } = require('express');
+import authRequest from "../middlewares/check-auth";
+
+const router = Router();
+
+const createR = controller.createRequest;
+const updateR = controller.updateRequest; 
+const deleteR = controller.deleteRequest;
+const getR = controller.getAllRequests;
```
Comment (Anguandia) [inline src/routes/requestRoute.js]: You could refactor here by destructuring individual controllers from controller, eg, 
`const {
createRequest,
updateRequest,
deleteRequest,
getAllRequests
} = controller`
Then use createRequest instead of createR, etc. But you may leave this as is for now, in the interest of time

File: src/tests/request.test.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+      .put(`/Request/${requestData.id}`)
+      .set('authorization', `Bearer ${token}`)
+      .send({
+        dateStart : "2021-01-08",
+        dateEnd : "2021-01-19" ,
+        idRoom : 1,
+      });
+    expect(res).to.have.status(200);
+  });
+});
```
Comment (Anguandia) [inline src/tests/request.test.js:133]: You have only tested the successful cases of CRUD operations, how sure are we that unviable requests will fail as must?
Additionally, please try to test other properties of expected responses than only the status code. This may sometimes be insufficient or outright misleading

**Review by Anguandia — CHANGES_REQUESTED** [review]
Good work. Please take note of the following:
- Always lint your work and enforce the agreed style guide before pushing
- Test non successful request cases too, especially requests that should return return responses within the 4XX status family
- Responses should have user-friendly messages instead of system-generated error messages and codes. This helps (1) the user understand and, (2) the translation system correctly display the response message in any of the supported languages

**Review by Anguandia — APPROVED** [review]
